### DISCUSSION TITLE
Scalability

### DISCUSSION THREAD

**Discussion by benzo8** [slskd/slskd#597] [Ideas]
As mentioned in one of my bug reports, I am sharing well over 150,000 tracks. There comes a point (most definitely if I leave the server overnight) when the scale of the /uploads page is such that it will no longer load, and before that point, if I click "Delete all succeeded", it will throw a tonne of error boxes before dropping out of the process with only some or none of the history cleared...

I'd love to see a setting the yaml for "Automatically delete completed/errored/cancelled" (separate options, I guess) to alleviate help with the scaleability of slskd - I don't really care who's downloaded what, and if I really did I could log it and scan the logs...

**Comment by jpdillingham** [comment]
Thanks for starting this discussion; users with high track counts have been on my mind from the start and this seems like a good place to solicit feedback.

I modeled the UI after official clients for two reasons; end-user familiarity, and the fact that I'm not a very imaginative UI designer.  This was a reasonable place to start, however it is becoming more and more evident that this design doesn't really work that well in a browser; there's a lot of data on the screen, it updates a lot, and the tabular nature doesn't really work well on mobile devices.

The upload interface needs to be redesigned but so far I haven't been able to figure out how it should look.  For performance reasons the amount of data on the screen at any given time needs to be greatly reduced, and as you've mentioned, cleanups shouldn't be such a pain.  If anyone has any suggestions I'd love to hear them, otherwise I'll think about mocking something up.

On the subject of scalability, share scanning is another pain point.  I've created #579 to broadly represent that effort, but I'm still thinking through all of the moving parts.

**Reply by benzo8** [reply]
So, I'm not a UI guy but I do have a lot of UX experience (UX-X?), which may be of use.

I do not know if I am a a "normal" user of SoulSeek or not. I do not use Lists and Groups very much, and I share a large number of individual files, which makes me think I'm probably not, so my wants/suggestions ought to be taken with a pinch of salt, but here goes:

TL: I think there needs to be a "overview" page of Uploads, into which you can drill down. This page should load at the same speed whether you've got 10 or 10,000 uploads. I imagine a histogram, showing upload amount against hours/days/weeks/months and then you drill down into each period by clicking on the bar. You can also clear a bar (or multiple bars) from the overview page by selecting (a checkbox or the) bar(s) and using a Clear _x_ button on the overview page - no need to drill down if you don't care to. You should be able to zoom between the periods before drilling down so that, for instance, if you have too many Uploads on a daily basis to load the page, you can go on an hourly basis.

Now, individual points, for the Uploads pages proper:

1. Master Collapse/Expand buttons
2. Collapse/Expand state shouldn't reset whenever the page is updated
3. Collapsed albums/users should have a Clear checkbox which affects everything under the collapse
4. As per 3, you should be able to select all items of a particular user in one go
5. Shift-click on checkboxes should select all boxes between the one clicked and the previous checked (as per Windows Explorer, et al)
6. Sorting options would be great: dated added, data completed, etc.

Hopefully this gives the discussion a good place to start from!

**Reply by jpdillingham** [reply]
I like the idea of an overview/dashboard page!  I think that would be a good place to start.
